### ISSUE TITLE
Adding a variable to define the context of the event

### ISSUE THREAD

**Issue by carolinverset** [hzi-braunschweig/SORMAS-Project#5014]
### Situation Description
In France, we import case and contact information because it is provided by the Health Insurance to the regions. To date, only cases and their contacts have been imported. But in a very short time we will also be receiving an event file. These events are classified according to specific categories that define the context of the event. French teams need to have a variable that can be used to import these 20 types of events or to manually define the context of an event.

### Feature Description
Given a classification very specific to France, and in order to allow all countries to use this new variable or not, we propose the following development:

-	create a "event context" variable in the form of a drop-down list
The modalities of this drop-down list can be defined independently for each server because they must be entered in an independent table via an sql script provided by the @ftavin team (cf. variants #4042)
This new variable will only appear in the GUI if modalities have been defined in the table mentioned above. If these terms are not defined, the variable will not be displayed in the application.
-	Insertion of this new variable in the events and actions directory as well as in the exports
-	Add a filter to restrict events according to this variable.

### Additional Information
@bernardsilenou @ftavin @emond-fuseau

**Comment by MateStrysewske** [comment]
@carolinverset What would be example values for this new field in France?

@ftavin Given that we have a lot of these things coming up, I'd like to hear your opinion on what's a better long-term approach with customisable things like these. I think adding a database table for each of them will bloat the data model pretty quickly. Do you think there's another approach - e.g. having configuration files instead - that we could use instead?

We might need to talk about this internally as well; I've not brought up that topic for the disease variants because they were incredibly urgent, but I'd like to find a solution that works for all these kinds of requests in the future.

**Comment by carolinverset** [comment]
> @carolinverset What would be example values for this new field in France?
> 
> @ftavin Given that we have a lot of these things coming up, I'd like to hear your opinion on what's a better long-term approach with customisable things like these. I think adding a database table for each of them will bloat the data model pretty quickly. Do you think there's another approach - e.g. having configuration files instead - that we could use instead?
> 
> We might need to talk about this internally as well; I've not brought up that topic for the disease variants because they were incredibly urgent, but I'd like to find a solution that works for all these kinds of requests in the future.


'EMES' => 'Medico-social establishments'
'FROP' => 'Home or boarding school'
'EPEN' => 'Prison establishment'
'HPJJ' => 'Reception and accommodation structure under the judicial protection of young people'
'TOUR' => 'Tourist accommodation structure'
'CRCH' => 'Childcare facility for young children'
'SCOL' => 'Event or situation in a school environment'
'UNIV' => 'Event or situation in a university or higher education establishment'
'ESPP' => 'Public or private health establishments'
'ACES' => 'Sporting activity or event'
'ACEC' => 'Cultural activity or event'
'RPFA' => 'Private meetings in a family, friendly or other setting'
'RAFP' => 'Festive gathering on public roads'
'RAEP' => 'Meeting or event related to professional activity'
'RCMP' => 'Collective catering in a professional environment'
'REST' => 'Restaurant'
'BARS' => 'Bars'
'PPTC' => 'Prolonged presence in public transport (> 15 min)'
'AUTR' => 'Other place / event without respecting barrier gestures'

**Comment by MateStrysewske** [comment]
@carolinverset At the first glance, this looks a lot like a mixture of the ExposureType and TypeOfPlace fields that are already part of SORMAS (e.g. "type of activity" - and the resulting detail fields, e.g. "type of gathering" - and "type of place" when creating an exposure in the epidemiological data section of a case or contact). It might make sense to think about a solution that allows us to use those already existing values and fields instead of creating a new one.

I have a discussion with two colleagues about the way we want to approach configurable enums in the system on Monday; we'll also quickly talk about this topic then.

**Comment by kwa20** [comment]
@MateStrysewske Good point! There have also been requests before to be able to migrate events to exposures when creating contacts or cases for event participants. This is still something that needs to be discussed because it would appear as a linked event and exposure which might not be ideal but I think consistency across those entities would be good.

**Comment by bernardsilenou** [comment]
@MateStrysewske @carolinverset 
 - "type of activity"  was the first idea that came to my mind when we discussed this topic yesterday, but we latter decided not to go this way because @carolinverset  said the users would need to see all the options in one variable. I think we can further think of refining this issue.
 - some of the categories fall in "type of place/ facility" and others in "type of activity". 
 - @carolinverset  If we can splitting this list in to 2 different variables, some goes to types of activity and others to type of place, then I can see a solution based on the current variables in the system. If we can not split, then we would we would need to go with a configurable solution as initially suggested.
 - One main point now is to decide the best way of configuring enums and even variables themselves on one instance only.
